src/components/PivotTable/PivotTableTitleRow.js

```diff
@@ -10,32 +11,62 @@ export const PivotTableTitleRow = ({
     containerWidth,
     totalWidth,
 }) => {
+    const containerRef = useRef(null)
+    const initialScrollWidthRef = useRef(null)
+    const handleContainerMount = useCallback((node) => {
+        if (node) {
+            containerRef.current = node
+            initialScrollWidthRef.current = node.scrollWidth
+        }
+    }, [])
```

Comment:
My initial implementation in commit https://github.com/dhis2/analytics/pull/1579/commits/b020edd88891b42407a0463eaa90629189c8c335 had a problem:
- To determine if the title is truncated I was checking if the container node's _current_ `scrollWidth` was greater than its current `clientWidth`.
- But in DV, the container width changes with a CSS transformation/animation, and during this animation the browser would sometimes report both values as equal even though the text doesn't fit
- The result was that sometimes after hiding/showing the details panel (which causes the canvas to shrink) the tooltip would not show, even though it should have.

So my solution revolves around the notion that (due to `white-space: nowrap` and  `overflow: hidden`) the initial `scrollWidth` of the element is correct and will remain constant. I think it all makes sense so far....

But the current implementation is not the easiest to follow and perhaps a bit quirky. I landed on it for the following reasons:
1. To measure a DOM node it [is recommended](https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node) to use a _callback ref_, which `handleContainerMount` is.
2. But we also need to read some attributes from the container element before the JSX is rendered, so we need a regular ref to it as well. But since we already attached a callback ref, we had to manually set that in the function body of the callback ref. 
3. And the same applies to the `initialScrollWidthRef`: we can set this value when the element mounts and trust it to remain constant.
4. Since both these things are effectively derived values, I did not want to store them in a state variable and possibly triggering additional renders. So that's why both are stored in a `ref`